src/main/java/rx/internal/operators/OperatorMaterialize.java

```diff
@@ -141,7 +138,7 @@ private void drain() {
                 Notification<T> tn;
                 tn = terminalNotification;
                 if (tn != null) {
-                    if (requested > 0) {
+                    if (requested.get() > 0) {
```

Comment:
The `requested` reference should be a local variable to avoid re-reading it in the loop as `get()` is a load-load barrier which forces all subsequent field accesses to happen fresh from LLC, including the `requested` reference itself.